test/unit/BCoWHelper.tree

```diff
@@ -0,0 +1,26 @@
+BCoWHelperTest::constructor
+└── when called
+    ├── it should set factory
+    └── it should set app data from factory
+
+BCoWHelperTest::tokens
+├── when pool is not registered in factory
+│   └── it should revert
+├── when pool has less than 2 tokens
+│   └── it should revert
+├── when pool has more than 2 tokens
+│   └── it should revert
+├── when pool tokens have different weights
+│   └── it should revert
+└── when pool is supported
+    └── it should return pool tokens
+
+BCoWHelperTest::order
+├── when the pool is not supported
```

Comment:
hmm, we cannot override a `public view` even if it's virtual :/
because we would need to do `address(this).call` within `tokens` to check it and that's not possible ona view method...

Reply:
mistaken, we **can** override `tokens`, added an assertion that is being called 👍

Reply:
if that's the case then we can now get rid of this test, right? :eyes: I'd say testing 'when one of the lines of this function reverts, the revert bubbles up' is something we can delegate to the compiler :grin: